Loosen dependency versions, add resiliency

WoosterInitiative opened the pull request:
All dependencies were pinned to specific versions which would fight with whatever project it was implemented in. Loosened them while trying to keep them reasonable.

Added a timeout for waiting for postgres to be ready.

WoosterInitiative committed:
chore: :arrow_up: Update pre-commit hooks and dependencies

WoosterInitiative committed:
fix(timeout): add timeout to postgres check

WoosterInitiative committed:
refactor(docker): new methods on PostgresEnvironment and ContainerConfig to be more explicit when creating environs

Commit:
adjust severel dep versions

emdoyle commented on bridge/service/postgres.py:
```diff
@@ -18,12 +21,29 @@ class PostgresEnvironment(BaseModel):
     name: str = "bridge_postgres"
     ports: dict[str, int] = {"5432/tcp": 5432}
-    volumes: dict[str, Union[list[str], dict[str, str]]] = Field(
+    volumes: dict[str, list[str] | dict[str, str]] = Field(
```
This syntax requires Python 3.10+ but we'd like to support 3.9 as well. I think adding `from __future__ import annotations` at the top of the file is probably enough to fix this!

emdoyle commented on bridge/service/postgres.py:
```diff
@@ -36,23 +56,26 @@ class PostgresConfig(ContainerConfig[PostgresEnvironment]):
-        )
+        dsn = self.config.environment.dsn
+        msg = f"DSN: {dsn}"
+        print(msg)
```
This should probably use `log_info`

emdoyle reviewed (approved):
Thanks for doing this! Once the 3.9 compatibility change is made this should be good to merge.

Commit:
build(postgres): add __future__ annotations

supports python 3.9

use log_info instead of print and updates a few pre-commit dependencies

WoosterInitiative commented:
OK, changes have been made.

I haven't been able to run tach on it, however. Seems to be an issue with pydantic? I don't have time right now to dig into that, but I also couldn't run tests on the tach package. I'm in Windows, fwiw.

WoosterInitiative commented:
This would also close #29

emdoyle commented:
> OK, changes have been made.
> 
> I haven't been able to run tach on it, however. Seems to be an issue with pydantic? I don't have time right now to dig into that, but I also couldn't run tests on the tach package. I'm in Windows, fwiw.

Apologies for the delay on this, but thanks! I think the `service/docker.py` file also needs the future statement but should be good to go otherwise.

The issue with Tach sounds like a known issue with backward compatibility in Tach. I'd check what version you are running and make sure it matches 0.1.2 (in dev-dependencies). If it does match, then I think ignore any failures for now since it will be a separate task to upgrade to the latest version of Tach anyway.

Commit:
fix(pydantic): support newer versions of pydantic and other helpful tweaks

Commit:
build(commitizen): add commitizen config in pyproject

Commit:
bump: version 0.1.3b0 → 0.1.3b1

Commit:
test(fixtures): improve fixtures to use a tmp_path

was checking actual folder locations for existence of `manage.py` and other files

Commit:
docs(faq): add instructions for specifying postgres version

change to code was made in 6de9fae056a18ba5943de3ea9b3054a30d51941b

Commit:
feat(postgres): version and image tag added as config options

Commit:
bump: version 0.1.3b1 → 0.2.0b0

WoosterInitiative committed:
build(deps): loosen constraints on deps for better compatibility
